test/wallet-test.js

```diff
@@ -2869,4 +2869,338 @@ describe('Wallet', function() {
+      assert.strictEqual(bal.confirmed, fund - (cTXCount * fee));
+      assert.strictEqual(bal.unconfirmed, fund - (uTXCount * fee));
+      assert.strictEqual(bal.ulocked, 0);
+      assert.strictEqual(bal.clocked, 0);
+    });
+
+    it('should send and confirm BID', async () => {
+      // Advance to bidding
+      wdb.height += network.names.treeInterval + 1;
+      const winningBid = await wallet.sendBid(name, value, lockup, {hardFee: fee});
+      uTXCount++;
+
+      // Check
+      let bal = await wallet.getBalance();
+      assert.strictEqual(bal.tx, 3);
+      assert.strictEqual(bal.coin, 3);
+      assert.strictEqual(bal.confirmed, fund - (cTXCount * fee));
+      assert.strictEqual(bal.unconfirmed, fund - (uTXCount * fee));
+      assert.strictEqual(bal.ulocked, lockup);
+      assert.strictEqual(bal.clocked, 0);
+
+      // Confirm BID
+      let block = {
+        height: wdb.height + 1,
+        hash: Buffer.alloc(32),
+        time: Date.now()
+      };
+      await wallet.txdb.add(winningBid, block);
+      cTXCount++;
+
+      const winningBlindFromMtx = winningBid.outputs
+        .find(o => o.covenant.isBid())
+        .covenant.getHash(3);
+      let allBids = await wallet.getBids();
+      assert.strictEqual(allBids.length, 1);
+      const winningBlindFromWallet = allBids.find(b =>
+        b.blind.equals(winningBlindFromMtx)
+      );
+      assert(winningBlindFromWallet);
+
+      bal = await wallet.getBalance();
+      assert.strictEqual(bal.tx, 3);
+      assert.strictEqual(bal.coin, 3);
+      assert.strictEqual(bal.confirmed, fund - (cTXCount * fee));
+      assert.strictEqual(bal.unconfirmed, fund - (uTXCount * fee));
+      assert.strictEqual(bal.ulocked, lockup);
+      assert.strictEqual(bal.clocked, lockup);
+
+      const auctionsTxs = await wallet.createAuctionTxs(name, secondHighest, lockup, {hardFee: fee, force:true});
+      const losingBidUnsent = auctionsTxs.bid;
+      unsentReveal = auctionsTxs.reveal;
+
+      const losingBid = await wallet.sendMTX(losingBidUnsent, null);
+      uTXCount++;
+
+      // Check
+      bal = await wallet.getBalance();
+      assert.strictEqual(bal.tx, 4);
+      assert.strictEqual(bal.coin, 4);
+      assert.strictEqual(bal.confirmed, fund - (cTXCount * fee));
+      assert.strictEqual(bal.unconfirmed, fund - (uTXCount * fee));
+      assert.strictEqual(bal.ulocked, 2 * lockup);
+      assert.strictEqual(bal.clocked, lockup);
+
+      // Confirm BID
+      block = {
+        height: wdb.height + 1,
+        hash: Buffer.alloc(32),
+        time: Date.now()
+      };
+      await wallet.txdb.add(losingBid, block);
+      cTXCount++;
+
+      const losingBlindFromMtx = losingBid.outputs
+        .find(o => o.covenant.isBid())
+        .covenant.getHash(3);
+      allBids = await wallet.getBids();
+      assert.strictEqual(allBids.length, 2);
+      const losingBlindFromWallet = allBids.find(
+        b => b.blind.toString('hex') === losingBlindFromMtx.toString('hex')
```

Comment:
can do that `Buffer.equals()` here again

Reply:
Changed !